src/CloudNative.CloudEvents/ICloudEventFormatter.cs

```diff
@@ -21,21 +20,26 @@ public interface ICloudEventFormatter
         /// <param name="data"></param>
         /// <param name="extensions"></param>
         /// <returns></returns>
-        CloudEvent DecodeStructuredEvent(Stream data, IEnumerable<ICloudEventExtension> extensions);
+        CloudEvent DecodeStructuredEvent(Stream data, IEnumerable<CloudEventAttribute> extensionAttributes);
+
         /// <summary>
         /// Decode a structured event from a stream asynchonously
         /// </summary>
         /// <param name="data"></param>
         /// <param name="extensions"></param>
         /// <returns></returns>
-        Task<CloudEvent> DecodeStructuredEventAsync(Stream data, IEnumerable<ICloudEventExtension> extensions);
+        Task<CloudEvent> DecodeStructuredEventAsync(Stream data, IEnumerable<CloudEventAttribute> extensionAttributes);
+
+        // TODO: Remove either this one or the stream one? It seems unnecessary to have both.
+
         /// <summary>
         /// Decode a structured event from a byte array
         /// </summary>
         /// <param name="data"></param>
         /// <param name="extensions"></param>
         /// <returns></returns>
-        CloudEvent DecodeStructuredEvent(byte[] data, IEnumerable<ICloudEventExtension> extensions);
+        CloudEvent DecodeStructuredEvent(byte[] data, IEnumerable<CloudEventAttribute> extensionAttributes);
```

Comment:
Should these use ROM<byte> instead of byte array?

Reply:
(Ack, need to have this discussion.)